client/homebrew/pages/vaultPage/vaultPage.jsx

```diff
@@ -0,0 +1,478 @@
+require('./vaultPage.less');
+
+const React = require('react');
+const { useState, useEffect, useRef } = React;
+const cx = require('classnames');
+
+const Nav = require('naturalcrit/nav/nav.jsx');
+const Navbar = require('../../navbar/navbar.jsx');
+const RecentNavItem = require('../../navbar/recent.navitem.jsx').both;
+const Account = require('../../navbar/account.navitem.jsx');
+const NewBrew = require('../../navbar/newbrew.navitem.jsx');
+const HelpNavItem = require('../../navbar/help.navitem.jsx');
+const BrewItem = require('../basePages/listPage/brewItem/brewItem.jsx');
+const SplitPane = require('../../../../shared/naturalcrit/splitPane/splitPane.jsx');
+
+const request = require('../../utils/request-middleware.js');
+
+const VaultPage = (props) => {
+	const [title, setTitle] = useState(props.query.title || '');
+	//state author
+	const [author, setAuthor] = useState(props.query.author || '');
+	const [legacy, setLegacy] = useState(props.query.legacy !== 'false');
+	const [v3, setV3] = useState(props.query.v3 !== 'false');
+	const [count, setCount] = useState(props.query.count || 20);
+	const [page, setPage] = useState(parseInt(props.query.page) || 1);
+	const [brewCollection, setBrewCollection] = useState(null);
+	const [totalBrews, setTotalBrews] = useState(null);
+	const [searching, setSearching] = useState(false);
+	const [error, setError] = useState(null);
+
+	const titleRef = useRef(null);
+	const authorRef = useRef(null);
+	const countRef = useRef(null);
+	const v3Ref = useRef(null);
+	const legacyRef = useRef(null);
+	const searchButtonRef = useRef(null);
+
+	useEffect(() => {
+		disableSubmitIfFormInvalid();
+		loadPage(page, false, true);
+	}, []);
+
+	const updateStateWithBrews = (brews, page) => {
+		setBrewCollection(brews || null);
+		setPage(parseInt(page) || 1);
+		setSearching(false);
+	};
+
+	const updateUrl = (title, author, count, v3, legacy, page) => {
+		const url = new URL(window.location.href);
+		const urlParams = new URLSearchParams();
+
+		Object.entries({
+			title,
+			author,
+			count,
+			v3,
+			legacy,
+			page,
+		}).forEach(([key, value]) => urlParams.set(key, value));
+
+		url.search = urlParams.toString();
+		window.history.replaceState(null, null, url);
+	};
+
+	const loadPage = async (page, update, total) => {
+		//Different searches use the update or total props to make only the necessary queries and functions
+
+		if (!validateForm()) {
+			return;
+		}
+
+		setSearching(true);
+		setError(null);
+
+		const performSearch = async ({ title, author, count, v3, legacy }) => {
```

Comment:
This is re-defining `performSearch` and `loadTotal` every time `loadPage` is called. These functions should be moved outside of `loadPage`.

Reply:
fixed